### PR TITLE
Small fixes to multi-repo checkout scenario

### PR DESCRIPTION
**Pull request by jpricket**
- Handle ConvertToLocalPath in agent code (this is used to make local paths relative to build dir)
 - Update checkout description to include name of repo and disk location - [user story 1629387](https://dev.azure.com/mseng/AzureDevOps/_workitems/edit/1629387)

### CODE REVIEW

File: src/Agent.Sdk/Util/RepositoryUtil.cs [hunk 1]
```diff
@@ -93,7 +135,7 @@ public static string GuessRepositoryType(string repositoryUrl)
 
                 return RepositoryTypes.Tfvc;
             }
-            else if (repositoryUrl.IndexOf("bitbucket.org", StringComparison.OrdinalIgnoreCase) >= 0)
+            else if (repositoryUrl?.IndexOf("bitbucket.org", StringComparison.OrdinalIgnoreCase) >= 0)
```
Comment (jtpetty) [inline src/Agent.Sdk/Util/RepositoryUtil.cs]: I missed this the first time, but should we early out up top if repositoryUrl is null or empty string?

File: src/Agent.Worker/Build/BuildJobExtension.cs [hunk 2]
```diff
@@ -152,85 +152,138 @@ public override void InitializeJobExtension(IExecutionContext executionContext,
...
-            // Prepare the build directory.
-            executionContext.Output(StringUtil.Loc("PrepareBuildDir"));
-            var directoryManager = HostContext.GetService<IBuildDirectoryManager>();
-            TrackingConfig trackingConfig = directoryManager.PrepareDirectory(
-                executionContext,
-                executionContext.Repositories,
-                workspace);
+            if (repoClean.HasValue)
+            {
+                executionContext.Variables.Set(Constants.Variables.Build.RepoClean, repoClean.Value.ToString());
```
Comment (jtpetty) [inline src/Agent.Worker/Build/BuildJobExtension.cs]: can you change this an the one on line 238 to use SetVariable ?

File: src/Agent.Sdk/Util/RepositoryUtil.cs (hunk 1 above)
Comment (jpricket) [reply src/Agent.Sdk/Util/RepositoryUtil.cs]: that's a good point. I will fix it.

File: src/Agent.Worker/Build/BuildJobExtension.cs (hunk 2 above)
Comment (jpricket) [reply src/Agent.Worker/Build/BuildJobExtension.cs]: Certainly!

**Pull request merged**